cpp/src/io/parquet/page_string_utils.cuh

```diff
@@ -129,16 +127,16 @@ __device__ void convert_small_string_lengths_to_offsets(page_state_s const* cons
  * @brief Atomically update the initial string offset to be used during large string column
  * construction
  */
+template <bool has_lists>
 inline __device__ void compute_initial_large_strings_offset(page_state_s const* const state,
-                                                            size_t& initial_str_offset,
-                                                            bool has_lists)
+                                                            size_t& initial_str_offset)
 {
   // Values decoded by this page.
   int value_count = state->nesting_info[state->col.max_nesting_depth - 1].value_count;
 
   // if no repetition we haven't calculated start/end bounds and instead just skipped
   // values until we reach first_row. account for that here.
-  if (not has_lists) { value_count -= state->first_row; }
+  if constexpr (not has_lists) { value_count -= state->first_row; }
```

Comment:
Is there a perf signal from this change? I'm surprised that we're generating another copy of `compute_initial_large_strings_offset` just to avoid a single branching. Overall I feel like I'm missing the reason for the addition of `has_lists` template param.

Reply:
Wash: https://github.com/rapidsai/cudf/pull/18347#issuecomment-2784931905